gapis/api/vulkan/frame_loop.go

```diff
@@ -126,129 +123,198 @@ func newFrameLoop(ctx context.Context, c *capture.GraphicsCapture, numInitialCmd
 			vkCmd := cmd.(*VkDestroyImage)
-			vkCmd.Extras().Observations().ApplyReads(s.Memory.ApplicationPool())
 			img := vkCmd.Image()
 			log.D(ctx, "Image %v destroyed", img)
 			f.imageDestroyed[img] = true
 
-		// TODO: Recreate destroyed resources.
-		default:
-			if err := cmd.Mutate(ctx, id, f.backupState, nil, f.watcher); err != nil {
-				return fmt.Errorf("%v: %v: %v", id, cmd, err)
-			}
+		// DescriptionSetLayout(s)
+		case *VkCreateDescriptorSetLayout:
+			// type VkCreateDescriptorSetLayout struct {
+			//         Thread               int64    `protobuf:"zigzag64,1,opt,name=thread,proto3" json:"thread,omitempty"`
+			//         Device               int64    `protobuf:"zigzag64,8,opt,name=device,proto3" json:"device,omitempty"`
+			//         PCreateInfo          int64    `protobuf:"zigzag64,9,opt,name=pCreateInfo,proto3" json:"pCreateInfo,omitempty"`
+			//         PAllocator           int64    `protobuf:"zigzag64,10,opt,name=pAllocator,proto3" json:"pAllocator,omitempty"`
+			//         PSetLayout           int64    `protobuf:"zigzag64,11,opt,name=pSetLayout,proto3" json:"pSetLayout,omitempty"`
+			//         XXX_NoUnkeyedLiteral struct{} `json:"-"`
+			//         XXX_unrecognized     []byte   `json:"-"`
+			//         XXX_sizecache        int32    `json:"-"`
+			// }
+			vkCmd := cmd.(*VkCreateDescriptorSetLayout)
+			descriptorSetLayout := vkCmd.PSetLayout().MustRead(ctx, vkCmd, f.backupState, nil)
+			log.D(ctx, "DescriptorSetLayout %v created", descriptorSetLayout)
+			f.descriptorSetLayoutCreated[descriptorSetLayout] = true
+
+		case *VkDestroyDescriptorSetLayout:
+			// type VkDestroyDescriptorSetLayout struct {
+			//         Thread               int64    `protobuf:"zigzag64,1,opt,name=thread,proto3" json:"thread,omitempty"`
+			//         Device               int64    `protobuf:"zigzag64,8,opt,name=device,proto3" json:"device,omitempty"`
+			//         DescriptorSetLayout  int64    `protobuf:"zigzag64,9,opt,name=descriptorSetLayout,proto3" json:"descriptorSetLayout,omitempty"`
+			//         PAllocator           int64    `protobuf:"zigzag64,10,opt,name=pAllocator,proto3" json:"pAllocator,omitempty"`
+			//         XXX_NoUnkeyedLiteral struct{} `json:"-"`
+			//         XXX_unrecognized     []byte   `json:"-"`
+			//         XXX_sizecache        int32    `json:"-"`
+			// }
+			vkCmd := cmd.(*VkDestroyDescriptorSetLayout)
+			descriptorSetLayout := vkCmd.DescriptorSetLayout()
+			log.D(ctx, "DescriptorSetLayout %v created", descriptorSetLayout)
+			f.descriptorSetLayoutDestroyed[descriptorSetLayout] = true
+
+		// DescriptorPool(s)
+		case *VkCreateDescriptorPool:
+			// vkCmd := cmd.(*VkCreateDescriptorPool)
+
+		case *VkDestroyDescriptorPool:
+			// vkCmd := cmd.(*VkDestroyDescriptorPool)
+
+		// DescriptorSet(s)
+		case *VkAllocateDescriptorSets:
+			// vkCmd := cmd.(*VkAllocateDescriptorSets)
+
+		case *VkFreeDescriptorSets:
+			// vkCmd := cmd.(*VkFreeDescriptorSets)
+
+		case *VkUpdateDescriptorSets:
```

Comment:
Not sure you should be doing anything on UpdateDescriptorSets. as discussed, we should diff all existing descriptor sets at the end, and then just update the ones that are different.